lib/hooks/core/hook-http.js

```diff
@@ -51,13 +51,13 @@ function requestWrap(request) {
 
     var namespace = cls.getNamespace();
     var uri = uriFromOptions(options);
-    var currContext = cls.getRootContext().createChildSpanData(uri);
-    currContext.span.setLabel(TraceLabels.HTTP_METHOD_LABEL_KEY, options.method);
-    currContext.span.setLabel(TraceLabels.HTTP_URL_LABEL_KEY, uri);
+    var span = agent.startSpan(uri);
+    span.addLabel(TraceLabels.HTTP_METHOD_LABEL_KEY, options.method);
```

Comment:
Can you pass the labels to the `agent.startSpan`? I think the goal should be to make sure `OpaqueSpan` is true to its name, or if that's not possible, call stop calling it `OpaqueSpan`.